activesupport/lib/active_support/core_ext/integer/time.rb

```diff
@@ -38,7 +38,7 @@ def months
   alias :month :months
 
   def years
-    ActiveSupport::Duration.new(self * 365.25.days, [[:years, self]])
+    ActiveSupport::Duration.new(self * 365.25.days.to_i, [[:years, self]])
```

Comment:
Should it be `self * 365.25.days.to_i`, or `(self * 365.25.days).to_i` ? If `self` is a fraction, the precision could be different.